### PR TITLE
[FIX] Map index stored as meta value in peptide identification 

### PR DESCRIPTION
**Pull request by lars20070**
The IDMapper can map peptide IDs to subelements of consensus features. But the information which subelement (i.e. which map) the ID was mapped to is lost.

With this pull request, the map index of the peptide feature which the ID was mapped to is stored as (optional) meta value in the peptide identification itself. This way it remains clear which subelement the ID was originally mapped to. The pull request solves #1216.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
+1 if passes

**Comment by timosachsenberg** [comment]
If you have time can you add a test that contains e.g. two Peptide ids at one consensus feature? Should be easy to construct from the idXML and the consensusXML.

**Comment by lars20070** [comment]
Ok. Will do.

**Comment by lars20070** [comment]
Now. - Two peptide IDs, same consensus subfeature.

**Comment by lars20070** [comment]
Now. - Two peptide IDs, same consensus subfeature.

**Comment by timosachsenberg** [comment]
maybe I overlooked something but I don't see multiple map indices annotated?

File: src/tests/class_tests/openms/data/IDMapper_3_out2.consensusXML [hunk 1]
```diff
@@ -1094,6 +1100,30 @@
...
+				</PeptideHit>
+				<PeptideHit score="-2.19199991226196" sequence="AHSD" charge="1" aa_before="R" aa_after="V" protein_refs="PH_169">
+				</PeptideHit>
+				<PeptideHit score="-2.2260000705719" sequence="ASHGG" charge="1" aa_before="V" aa_after="T" protein_refs="PH_170">
+				</PeptideHit>
+				<PeptideHit score="-2.23600006103516" sequence="ASSY" charge="1" aa_before="I" aa_after="F" protein_refs="PH_171">
+				</PeptideHit>
+				<PeptideHit score="-2.26500010490417" sequence="APAGN" charge="1" aa_before="N" aa_after="V" protein_refs="PH_172">
+				</PeptideHit>
+				<userParam type="int" name="map index" value="3"/>
```
Comment (timosachsenberg) [inline src/tests/class_tests/openms/data/IDMapper_3_out2.consensusXML]: ah here it is

**Comment by lars20070** [comment]
Nope. You wanted two IDs same feature (from map 3). So both IDs get "map index"=3.

Guess you want two subfeatures, same ID.

**Comment by timosachsenberg** [comment]
I actually did not say in detail what I want :) Maybe add another one at e.g. map index 2?

File: src/openms/include/OpenMS/ANALYSIS/ID/IDMapper.h [hunk 2]
```diff
@@ -259,10 +258,11 @@ namespace OpenMS
       @param ids PeptideIdentification for the ConsensusFeatures
       @param protein_ids ProteinIdentification for the ConsensusMap
       @param measure_from_subelements Do distance estimate from FeatureHandles instead of Centroid
+      @param annotate_ids_with_subelements Store map index of FeatureHandle in peptide identification?
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/ANALYSIS/ID/IDMapper.h]: why ?

**Pull request merged**